spec/testing/context_factory_spec.rb

```diff
@@ -30,11 +32,27 @@
     it 'creates a context up-to the action defined' do
       ctx =
         LightService::Testing::ContextFactory
-        .make_from(TestDoubles::AdditionOrganizer)
+        .make_from(organizer)
         .for(TestDoubles::AddsThreeAction)
-        .with(:number => 1)
+        .with(1)
 
       expect(ctx.number).to eq(4)
     end
   end
+
+  context 'when called with an invalid action' do
+    let(:invalid_action) { TestDoubles::MakesLatteAction }
+
+    xit 'raises an argument error' do
```

Comment:
There's not a way to do this currently with the ContextFactory using `before_action`s

Reply:
By introducing a bit more state in the `ContextFactory` we could make this pass:

```ruby
module LightService
  module Testing
    class ContextFactory
      attr_reader :organizer

      def self.make_from(organizer)
        new(organizer)
      end

      def for(action)
        @action_matched = false

        @organizer.before_actions = [
          lambda do |ctx|
            if ctx.current_action == action
              @action_matched = true
              throw(:return_ctx_from_execution, ctx)
            end
          end
        ]

        self
      end

      def with(ctx)
        escaped = catch(:return_ctx_from_execution) do
          @organizer.call(ctx)
        end

        if @action_matched == false
          raise ArgumentError.new "Could not find action in #{self.organizer}"
        end

        escaped
      end

      def initialize(organizer)
        @organizer = organizer
      end
    end
  end
end
```

However, thanks to your testing, [this spec](https://github.com/jeremy-hanna/light-service/blob/380290964c33e1fec6045351ecd98af56add152c/spec/testing/context_factory/reduce_if_spec.rb#L30-L38) fails.

Is this something we could tweak here?

I don't want to have pending specs in LightService and not having a solution for it.

Reply:
If you want to add state to this class for this functionality, please feel free to do so as an append to this PR or in another - I was only mentioning something that my solution would not achieve and possibly indicate a passing context to the user (developer) when in reality the organizer executed the entire action set which was not expected